tests/markov/msm/test_msm.py

```diff
@@ -55,167 +56,144 @@ def estimate_markov_model(dtrajs, lag, return_estimator=False, **kw) -> MarkovSt
-        q[1:] = 0.5
-        p[0:-1] = 0.5
-        q[2] = 1.0 - 10 ** (-b)
-        q[4] = 10 ** (-b)
-        p[2] = 10 ** (-b)
-        p[4] = 1.0 - 10 ** (-b)
-
-        bdc = BirthDeathChain(q, p)
-        P = bdc.transition_matrix()
-        cls.dtraj = generate_traj(P, 10000, start=0)
-        cls.tau = 1
-
-        """Estimate MSM"""
-        import inspect
-        argspec = inspect.getfullargspec(MaximumLikelihoodMSM)
-        default_maxerr = argspec.defaults[argspec.args.index('maxerr') - 1]
-        cls.C_MSM = msmest.count_matrix(cls.dtraj, cls.tau, sliding=True)
-        cls.lcc_MSM = msmest.largest_connected_set(cls.C_MSM)
-        cls.Ccc_MSM = msmest.largest_connected_submatrix(cls.C_MSM, lcc=cls.lcc_MSM)
-        cls.P_MSM = msmest.transition_matrix(cls.Ccc_MSM, reversible=True, maxerr=default_maxerr)
-        cls.mu_MSM = msmana.stationary_distribution(cls.P_MSM)
-        cls.k = 3
-        cls.ts = msmana.timescales(cls.P_MSM, k=cls.k, tau=cls.tau)
-
-    @classmethod
-    def tearDownClass(cls) -> None:
-        """Revert the state of the rng"""
-        np.random.mtrand.set_state(cls.state)
-
-    def test_MSM(self):
-        msm = estimate_markov_model(self.dtraj, self.tau)
-        self.assertEqual(self.tau, msm.count_model.lagtime)
-        assert_allclose(self.lcc_MSM, msm.count_model.connected_sets()[0])
-        self.assertTrue(np.allclose(self.Ccc_MSM.toarray(), msm.count_model.count_matrix))
-        self.assertTrue(np.allclose(self.C_MSM.toarray(), msm.count_model.count_matrix))
-        self.assertTrue(np.allclose(self.P_MSM.toarray(), msm.transition_matrix))
-        assert_allclose(self.mu_MSM, msm.stationary_distribution)
-        assert_allclose(self.ts[1:], msm.timescales(self.k - 1))
-
-    def test_MSM_sparse(self):
-        msm = estimate_markov_model(self.dtraj, self.tau, sparse=True)
-        self.assertEqual(self.tau, msm.count_model.lagtime)
-        assert_allclose(self.lcc_MSM, msm.count_model.connected_sets()[0])
-        self.assertTrue(np.allclose(self.Ccc_MSM.toarray(), msm.count_model.count_matrix.toarray()))
-        self.assertTrue(np.allclose(self.C_MSM.toarray(), msm.count_model.count_matrix.toarray()))
-        self.assertTrue(np.allclose(self.P_MSM.toarray(), msm.transition_matrix.toarray()))
-        assert_allclose(self.mu_MSM, msm.stationary_distribution)
-        assert_allclose(self.ts[1:], msm.timescales(self.k - 1))
+@pytest.mark.parametrize("reversible", [True, False])
+@pytest.mark.parametrize("statdist", [None, np.array([0.5, 0.5]), np.array([1.1, .5]), np.array([.1, .1]),
+                                      np.array([-.1, .5])])
+@pytest.mark.parametrize("sparse", [True, False])
+@pytest.mark.parametrize("maxiter", [1])
+@pytest.mark.parametrize("maxerr", [1e-3])
+def test_estimator_params(reversible, statdist, sparse, maxiter, maxerr):
+    if statdist is not None and (np.any(statdist > 1) or np.any(statdist < 0)):
+        with np.testing.assert_raises(ValueError):
```

Comment:
is there a reason that sometimes you call from `np.testing` when you've also included the import `from numpy.testing import *`?

Reply:
an oversight, forgot to do the final grep :)